src/rgw/rgw_bucket.cc

```diff
@@ -1470,7 +1470,8 @@ int RGWBucketAdminOp::limit_check(RGWRados *store,
 	}
 
 	num_shards = info.num_shards;
-	uint64_t objs_per_shard = num_objects / num_shards;
+	uint64_t objs_per_shard = (num_shards) ?
+	  num_objects/num_shards : std::max(uint64_t(0), num_objects);
```

Comment:
`num_objects` is unsigned, so `max(0, num_objects)` is a noop. looks good otherwise